### PR TITLE
chore(examples): add dontpanic controller

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit showcases how metac controller behaves when CRD is not applied against the k8s cluster but the controller tries to use an instance of this CRD.

It is expected that the controller should not panic immediately & should provide root cause of the error. This is helpful since there can be cases where a CRD is missing but controllers try to use their instances without above knowledge.

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

### CODE REVIEW

File: examples/gctl/dont-panic/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+					// with its corresponding attachment
+					"dontpanic/uid": string(request.Watch.GetUID()),
+				},
+			},
+			"spec": map[string]interface{}{
+				"message": "My CRD is not installed!",
+			},
+		},
+	}
+	// add this desired Pod instance to response
```
Comment (luisdavim) [inline examples/gctl/dont-panic/cmd/main.go:60]: nit: `object` not `pod`

Comment (AmitKumarDas) [reply examples/gctl/dont-panic/cmd/main.go]: updated

File: examples/gctl/dont-panic/README.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+```
+
+```sh
+# install namespace, rbac, crds & operator
+sudo kubectl apply -f dontpanic-ns.yaml
+sudo kubectl apply -f dontpanic-rbac-crd.yaml
+sudo kubectl apply -f dontpanic-operator.yaml
+
+# verify if above were installed properly
+sudo kubectl get ns
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply examples/gctl/dont-panic/README.md:74]: mentioned the reason

File: examples/gctl/dont-panic/README.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
+## DontPanic K8s controller
+
+- This is a K8s controller that imports metac as library
+  - In other words, this uses inline hooks than webhooks
+- Controller's K8s dependencies is configured in a config file
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply examples/gctl/dont-panic/README.md:5]: ok

File: examples/gctl/dont-panic/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
+## DontPanic K8s controller
+
+- This is a K8s controller that imports metac as library
+  - In other words, this uses inline hooks than webhooks
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply examples/gctl/dont-panic/README.md:4]: ok

File: examples/gctl/dont-panic/README.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  - Refer: config/config.yaml
+- Controller business logic is implemented in Go
+  - Refer: cmd/main.go
+  - K8s is completely abstracted from this logic
+  - Logic is implemented in respective reconcile functions
+    - A CR `IAmError` is sent as the response i.e. desired state
+    - NOTE: `IAmError`'s definition i.e. _(CRD)_ is not set in k8s cluster
+    - Hence this reconciliation should **error out**
+- `Expectations`:
+    - However, binary should **not panic**
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply examples/gctl/dont-panic/README.md:15]: ok

File: examples/gctl/dont-panic/README.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+## DontPanic K8s controller
+
+- This is a K8s controller that imports metac as library
+  - In other words, this uses inline hooks than webhooks
+- Controller's K8s dependencies is configured in a config file
+  - Refer: config/config.yaml
+- Controller business logic is implemented in Go
+  - Refer: cmd/main.go
+  - K8s is completely abstracted from this logic
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply examples/gctl/dont-panic/README.md]: ok

File: examples/gctl/dont-panic/cmd/main.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+// request. Similarly, SyncHookResponse is the payload sent as a
+// response as part of reconcile request.
+//
+// NOTE:
+//	SyncHookRequest has the resource that is under watch (here
+// DontPanic custom resource)
+//
+// NOTE:
+//  Both SyncHookRequest & SyncHookResponse have the resources that
+// form the desired state w.r.t the watched resource. These desired
```
[the remark this answers is not in the document]
Comment (AmitKumarDas) [reply examples/gctl/dont-panic/cmd/main.go]: modified to make it more readable

**Comment by AmitKumarDas** [comment]
@grzesuav Have updated with the requested changes.
Added same changes to hello-world examples as well.

**Pull request merged**